### PR TITLE
Migrate permission target to framework

### PR DESCRIPTION
**Pull request by alexhung**
* Migrate artifactory_permission_target to plugin framework
* Remove deprecated `artifactory_permission_targets` resource from SDKv2 provider
* DRY the error handling function and move to shared module

### CODE REVIEW

**Review by shrajfr12 — DISMISSED** [review]
lgtm

File: pkg/artifactory/resource/security/resource_artifactory_permission_target.go [hunk 1]
```diff
@@ -47,329 +240,266 @@ type Actions struct {
...
-	buildSchema := principalSchema
-	buildSchema.Elem.(*schema.Resource).Schema["repositories"].Description = `This can only be 1 value: "artifactory-build-info", and currently, validation of sets/lists is not allowed. Artifactory will reject the request if you change this`
-
-	return map[string]*schema.Schema{
-		"name": {
-			Type:     schema.TypeString,
-			Required: true,
-			ForceNew: true,
+		Validators: []validator.Object{
+			validatorfw.RequireIfDefined(path.Expressions{ // use customer validator to ensure 'repositories' attribute is set if this block is defined in configuration. See https://github.com/hashicorp/terraform-plugin-framework/issues/740
```
Comment (danielmkn) [inline pkg/artifactory/resource/security/resource_artifactory_permission_target.go:306]: customer -> custom

File: pkg/artifactory/resource/security/resource_artifactory_permission_target.go [hunk 2]
```diff
@@ -47,329 +240,266 @@ type Actions struct {
...
-func resourcePermissionTargetRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	permissionTarget := new(PermissionTargetParams)
-	resp, err := m.(utilsdk.ProvderMetadata).Client.R().SetResult(permissionTarget).Get(PermissionsEndPoint + d.Id())
 	if err != nil {
-		if resp != nil && resp.StatusCode() == http.StatusNotFound {
-			d.SetId("")
-			return nil
-		}
-		return diag.FromErr(err)
+		resp.Diagnostics.AddError(
```
Comment (danielmkn) [inline pkg/artifactory/resource/security/resource_artifactory_permission_target.go:467]: You can replace it with `utilfw.UnableToDeleteResourceError(resp, err)`

File: pkg/artifactory/resource/security/resource_artifactory_permission_target.go [hunk 3]
```diff
@@ -47,329 +240,266 @@ type Actions struct {
...
 	}
 
-	return PackPermissionTarget(permissionTarget, d)
-}
-
-func resourcePermissionTargetUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	permissionTarget := unpackPermissionTarget(d)
+	// Return error if the HTTP status code is not 200 OK or 404 Not Found
+	if response.StatusCode() != http.StatusNotFound && response.StatusCode() != http.StatusOK {
+		resp.Diagnostics.AddError(
```
Comment (danielmkn) [inline pkg/artifactory/resource/security/resource_artifactory_permission_target.go:479]: Same here.

**Review by danielmkn — CHANGES_REQUESTED** [review]
Minor changes required.

**Review by danielmkn — CHANGES_REQUESTED** [review]
Minor changes required.

File: pkg/artifactory/provider/framework.go [hunk 4]
```diff
@@ -164,11 +164,12 @@ func (p *ArtifactoryProvider) Configure(ctx context.Context, req provider.Config
 // Resources satisfies the provider.Provider interface for ArtifactoryProvider.
 func (p *ArtifactoryProvider) Resources(ctx context.Context) []func() resource.Resource {
 	return []func() resource.Resource{
-		user.NewArtifactoryUserResource,
-		user.NewArtifactoryManagedUserResource,
-		user.NewArtifactoryAnonymousUserResource,
-		security.NewArtifactoryGroupResource,
-		security.NewArtifactoryScopedTokenResource,
+		user.NewUserResource,
```
[the remark this answers is not in the document]
Comment (alexhung) [reply pkg/artifactory/provider/framework.go:167]: Sure. I didn't remove since this is a func to create new resource. So that kinda make sense.

**Review by danielmkn — APPROVED** [review]
Minor changes required.

**Pull request merged**